File: mealprep-api/Models/Meal.cs
namespace MealPrepAPI.Models
{
    public class Meal
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public string? CreatedBy { get; set; }
        public string FoodsJson { get; set; }
        public string GroupsJson { get; set; }
        public bool IsShared { get; set; }
        public string? Notes { get; set; }
        public string? PeopleJson { get; set; }
        public int? PeopleCount { get; set; }
        public int? TotalMeals { get; set; }
        public double? TotalCalories { get; set; }
    }
}

File: mealprep-web/src/types/meal.ts
// interface for a saved meal object from the database
export interface Meal {
  id: number;           // unique meal ID (autoincrement from backend)
  name: string;         // user-given name for the meal
  createdAt: string;    // timestamp of when the meal was saved
  createdBy?: string;   // optional user or device ID that saved the meal
  foodsJson: string;    // stringified array of food objects in the meal
  groupsJson: string;   // stringified array of ingredient groups
  isShared: boolean;    // whether the meal was shared (via link or code)
  notes?: string;       // optional notes for a meal
  peopleJson?: string;  // people information for calories per meal
  peopleCount?: number; // amount of people the meal prep was for
  totalMeals?: number;  // total meals the meal prep made
  totalCalories?: number; // total calories for entire meal prep
}